plftool doesn't compile on ubuntu 11.10

```
gcc -std=gnu99 -O2 -Wall -Werror -I../libplf  -ggdb -o plftool.o -c plftool.c 
plftool.c: In function ‘write_file’:
plftool.c:149:13: error: variable ‘retval’ set but not used 
[-Werror=unused-but-set-variable]
cc1: all warnings being treated as errors

gcc -std=gnu99 -O2 -Wall -Werror -I../libplf -ggdb -o ini.o ini.c
ini.c: In function ‘ini_open’:
ini.c:55:12: error: variable ‘read_bytes’ set but not used 
[-Werror=unused-but-set-variable]
cc1: all warnings being treated as errors

Patch attached
```

Original issue reported on code.google.com by `Ilya.Nov...@gmail.com` on 30 Jan 2012 at 1:44

Attachments:
- [plftool.diff](https://storage.googleapis.com/google-code-attachments/ardrone-tool/issue-3/comment-0/plftool.diff)

--- a/projects/Makefile
+++ b/projects/Makefile
@@ -1,0 +1,7 @@
+PROJECTS=libplf plfinfo plf_inst_extract plftool usb_flash
+.PHONY:	all
+
+all:
+	for project in $(PROJECTS); do \
+		$(MAKE) -C "$$project" ; \
+	done

--- a/projects/plf_inst_extract/.gitignore
+++ b/projects/plf_inst_extract/.gitignore
@@ -1,0 +1,1 @@
+plf_inst_extract

--- a/projects/plfinfo/.gitignore
+++ b/projects/plfinfo/.gitignore
@@ -1,0 +1,1 @@
+plfinfo

--- a/projects/plfinfo/plfinfo.c
+++ b/projects/plfinfo/plfinfo.c
@@ -171,7 +171,7 @@ FileInfo_exit_1:
 
 void print_help(const char* name)
 {
-    printf("usage: %s [-h] [-o <output_file>] -i <updater_file>\n", name);
+    printf("usage: %s [-h] [-s] [-v] -i <updater_file>\n", name);
 
     printf("%-40s %s\n", "-h, --help",                      "Print this information");
     printf("%-40s %s\n", "-i, --input-file <input_file>",   "Input file name");
@@ -190,7 +190,7 @@ int parse_options(int argc, char** argv)
     while(1)
     {
         int option_index;
-        int result = getopt_long(argc, argv, "i:h:s:v", long_options, &option_index);
+        int result = getopt_long(argc, argv, "i:hsv", long_options, &option_index);
 
         if (result < 0)
             return 0;

--- a/projects/plftool/plftool.c
+++ b/projects/plftool/plftool.c
@@ -157,7 +157,6 @@ int write_file(const char* path, const void* data, u32 len, u32 umask)
 #else
     fi = open(path,  O_WRONLY | O_CREAT, umask);
 #endif
-    retval = 0;
 
     if (fi >= 0)
     {
@@ -166,7 +165,7 @@ int write_file(const char* path, const void* data, u32 len, u32 umask)
     }
     else
     {
-        return -1;
+        retval = -1;
     }
     return retval;
 }

--- a/projects/usb_flash/.gitignore
+++ b/projects/usb_flash/.gitignore
@@ -1,0 +1,1 @@
+usb_flash

